crates/bevy_render/src/texture/texture_attachment.rs

```diff
@@ -120,3 +120,41 @@ impl DepthAttachment {
         }
     }
 }
+
+/// A wrapper for a [`TextureView`] that is used as a [`RenderPassColorAttachment`] for a view
+/// targets final output texture.
+#[derive(Clone)]
+pub struct OutputColorAttachment {
+    pub view: TextureView,
+    pub format: TextureFormat,
+    is_first_call: Arc<AtomicBool>,
+}
+
+impl OutputColorAttachment {
+    pub fn new(view: TextureView, format: TextureFormat) -> Self {
+        Self {
+            view,
+            format,
+            is_first_call: Arc::new(AtomicBool::new(true)),
+        }
+    }
+
+    /// Get this texture view as an attachment. The attachment will be cleared with a value of
+    /// the provided `clear_color` if this is the first time calling this function, otherwise it
+    /// will be loaded.
+    pub fn get_attachment(&self, clear_color: Option<LinearRgba>) -> RenderPassColorAttachment {
```

Comment:
I'm not a super fan of this pattern which consists in making a getter function an implicit mutator. This is a footgun. At the very least if we don't want to change the pattern, I'd suggest to change the name of the function to make it clear it mutates the object, like `use_as_attachment()`.

Reply:
The mutator doesn't really matter, when the texture gets cleared is an implementation detail. The existing types in this file all follow this pattern. But anyways, it's not a huge deal either way. This is all a temporary fix until #13397 lands.

Reply:
I agree passing an argument to a mutator isn't great, but this is again just trying to follow similar conventions to the rest of the file, for example `DepthAttachment` which accepts a `StoreOp`. This could definitely all be cleaned up in another PR.